Sources/Models/MessageStyle.swift

```diff
@@ -79,31 +79,40 @@ public enum MessageStyle {
 
         let cache = MessageStyle.bubbleImageCache
 
-        if let cachedImage = cache.object(forKey: path as NSString) {
-            return cachedImage
-        }
-
-        guard var image = UIImage(contentsOfFile: path) else { return nil }
-
         switch self {
         case .none, .custom:
             return nil
         case .bubble, .bubbleOutline:
-            break
+            if let cachedImage = cache.object(forKey: path as NSString) {
```

Comment:
Rather than breaking out cache miss logic per-case, why not add a `cacheKey` computed property that returns `path` or `path + corner.rawValue` depending on the case and use that? The only thing different between these cases is the cache key so it would be more terse/DRY that way.

Reply:
Good idea 👍 .